packages/amplify-graphql-api-construct/API.md

```diff
@@ -0,0 +1,145 @@
+## API Report File for "@aws-amplify/graphql-api-construct-alpha"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { CfnApiKey } from 'aws-cdk-lib/aws-appsync';
+import { CfnDataSource } from 'aws-cdk-lib/aws-appsync';
+import { CfnFunctionConfiguration } from 'aws-cdk-lib/aws-appsync';
+import { CfnGraphQLApi } from 'aws-cdk-lib/aws-appsync';
+import { CfnGraphQLSchema } from 'aws-cdk-lib/aws-appsync';
+import { CfnPolicy } from 'aws-cdk-lib/aws-iam';
+import { CfnResolver } from 'aws-cdk-lib/aws-appsync';
+import { CfnResource } from 'aws-cdk-lib';
+import { CfnRole } from 'aws-cdk-lib/aws-iam';
+import { CfnTable } from 'aws-cdk-lib/aws-dynamodb';
+import { Construct } from 'constructs';
+import { Duration } from 'aws-cdk-lib';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { IFunction } from 'aws-cdk-lib/aws-lambda';
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { IUserPool } from 'aws-cdk-lib/aws-cognito';
+import { ResolverConfig } from '@aws-amplify/graphql-transformer-core';
+import { SchemaFile } from 'aws-cdk-lib/aws-appsync';
+import { TransformerPluginProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { TransformParameters } from '@aws-amplify/graphql-transformer-interfaces';
+
+// @public (undocumented)
+export class AmplifyGraphQlApi extends Construct {
+    constructor(scope: Construct, id: string, props: AmplifyGraphQlApiProps);
+    // (undocumented)
+    readonly env: string;
+    // (undocumented)
+    readonly resources: AmplifyGraphQlApiResources;
+}
+
+// @public (undocumented)
+export type AmplifyGraphQlApiProps = {
```

Comment:
I know we had a brief conversation about how this construct is going to reference lambdas for @function resolvers, but I'm still a bit nervous about this integration point. Maybe this is something that we can iterate on over time, but I think this construct is going to need to consume something like a FunctionLocator with an interface like
```ts
type FunctionLocator = {
  locateFunction(name: string): IFunction
}
```
When resolving the directive `@function(name: "myCoolResolver")`, the transformer would call `functionLoader.locateFunction('myCoolResolver')` and then configure the returned function as the DataSource for that field.

If a prop for this in the `AmplifyGraphQlApiProps` is undesireable, we could also have a method on the construct like
```
withFunctionLocator(locator: FunctionLocator): void
```
that gets injected outside of the ctor. This may be more desirable because the FunctionLocator is only needed if the schema has @function directives. Of course this comes with other problems like how to wire in those functions after the bulk of the transformation already happened in the ctor. But maybe this

Reply:
Yeah, providing a function name to IFunction or locator as you're describing is in the plan, but until we're no longer internally synthesizing the CDK out to cloudassembly then reimporting we can't pass those cdk-resolved tokens over the app barrier. We'll discuss in the API review, but the intent is to have some affordance like you're describing, it just won't work today.